### PR TITLE
docs(schemagen): Update/expand documentation on propertynaming

### PR DESCRIPTION
**Pull request by ProbablePrime**
It seems this area of the documentation was a little out of date, I have updated it!

### CODE REVIEW

**Comment by ProbablePrime** [comment]
@dotnet-policy-service agree

File: _docs/schema/schemagen/schema-generation.md [hunk 1]
```diff
@@ -270,16 +270,34 @@ There are four options:
 
 ### Property naming {#schema-schemagen-property-names}
 
-In addition to the `[JsonPropertyName]` attribute, the configuration exposes a `PropertyNamingMethod` that allows you to define a custom method for altering property names from your code into the schema.  The `PropertyNamingMethods` static class defines a few commonly-used conventions:
+In addition to the `[JsonPropertyName]` attribute, the configuration(`SchemaGeneratorConfiguration`) exposes a `PropertyNameResolver` delegate that allows you to define a custom method for altering property names from your code into the schema. The system will adjust property names accordingly.
```
Comment (gregsdennis) [inline _docs/schema/schemagen/schema-generation.md]:
```suggestion
In addition to the `[JsonPropertyName]` attribute, `SchemaGeneratorConfiguration.PropertyNameResolver` allows you to define a custom method for altering property names from your code into the schema. The system will adjust property names accordingly.
```

File: _docs/schema/schemagen/schema-generation.md [hunk 2]
```diff
@@ -270,16 +270,34 @@ There are four options:
...
-- `UPPER_SNAKE_CASE`
+```C#
+SchemaGeneratorConfiguration config = new()
+{
+  // All property names will be lowercase!
+  PropertyNameResolver = x => x.Name.ToLower()
+};
+```
+
+The `PropertyNameResolvers` static class defines a few commonly-used conventions:
```
Comment (gregsdennis) [inline _docs/schema/schemagen/schema-generation.md]:
```suggestion
For your convenience, the `PropertyNameResolvers` static class defines a few commonly-used conventions:
```

File: _docs/schema/schemagen/schema-generation.md [hunk 3]
```diff
@@ -270,16 +270,34 @@ There are four options:
...
+|----------------------------------------|---------------------|
+| PropertyNameResolvers.CamelCase        | `camelCase`         |
+| PropertyNameResolvers.PascalCase       | `PascalCase`        |
+| PropertyNameResolvers.KebabCase        | `kebab-case`        |
+| PropertyNameResolvers.UpperKebabCase   | `UPPER-KEBAB-CASE`  |
+| PropertyNameResolvers.SnakeCase        | `snake_case`        |
+| PropertyNameResolvers.UpperSnakeCase   | `UPPER_SNAKE_CASE`  |
 
-Just set this property and the system will adjust property names accordingly.
+You can use them like this:
```
Comment (gregsdennis) [inline _docs/schema/schemagen/schema-generation.md]:
```suggestion
They can be applied directly to the configuration property:

```

**Review by gregsdennis — CHANGES_REQUESTED** [review]
Thanks for catching this, please see the recommendations, but I think it's good overall.

**Comment by ProbablePrime** [comment]
Applied all your suggestions, thank you!

**Review by gregsdennis — APPROVED** [review]

**Pull request merged**